### PR TITLE
solve interval panic

### PR DESCRIPTION
**Pull request by bbbearxyz**
**What type of PR is this?**

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

issue #3022

**What this PR does / why we need it:**

Not Available

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

File: pkg/container/types/date.go [hunk 1]
```diff
@@ -63,7 +63,7 @@ var (
 
 const (
 	MaxDateYear    = 9999
-	MinDateYear    = 0
+	MinDateYear    = 1000
```
Comment (broccoliSpicy) [inline pkg/container/types/date.go:66]: what's wrong with the original range?

**Comment by broccoliSpicy** [comment]
is there a way to fix issue#3022 without change our datetime's original range?

File: pkg/container/types/datetime.go [hunk 2]
```diff
@@ -29,6 +29,8 @@ const (
 	secsPerDay    = 24 * secsPerHour
 	//secsPerWeek   = 7 * secsPerDay
 	microSecondBitMask = 0xfffff
+	MaxDatetimeYear    = 2105
```
[the remark this answers is not in the document]
Comment (JinHai-CN) [reply pkg/container/types/datetime.go:32]: the range for DATETIME values is '1000-01-01 00:00:00.000000' to '9999-12-31 23:59:59.999999', and the range for TIMESTAMP values is '1970-01-01 00:00:01.000000' to '2038-01-19 03:14:07.999999'.

**Comment by JinHai-CN** [comment]
> is there a way to fix issue#3022 without change our datetime's original range?

What is original range of DATETIME ? 0000~9999?

File: pkg/container/types/datetime.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (JinHai-CN) [reply pkg/container/types/datetime.go:32]: > is there a way to fix issue#3022 without change our datetime's original range?

https://docs.microsoft.com/zh-cn/sql/t-sql/data-types/datetime-transact-sql?view=sql-server-ver15
In SQLServer, it is 1753~9999.

**Comment by broccoliSpicy** [comment]
> > > > is there a way to fix issue#3022 without change our datetime's original range?
> > > 
> > > 
> > > What is original range of DATETIME ? 0000~9999?
> > 
> > 
> > > > is there a way to fix issue#3022 without change our datetime's original range?
> > > 
> > > 
> > > What is original range of DATETIME ? 0000~9999?
> > 
> > 
> > yes
> 
> Check my previous reply. I would like to say MySQL datetime range is much larger than SQL-Server. :)

the original datetime range in matrixone is probably defined by @aunjgr, and the reason for this range 0-9999 is perhaps just for implementation convenience, for example, easier for leap year calculation

**Comment by aunjgr** [comment]
Actually I can't remember I set the year upper bound as 9999. Maybe it's for convenience of parsing?

I used an unsigned integer to store the sec/microsec elapsed from 0001-01-01 00:00:00.000000. The highest 44 bits for secs and lowest 20 bits for microsecs. So 1 is a legal year number but 0 is not.

**Comment by yingfeng** [comment]
Still panic lefts:
```
select date_sub(NULL,INTERVAL 100000 SECOND);

[EXPECT RESULT]:
date_sub(NULL,INTERVAL 100000 SECOND)
null
[ACTUAL RESULT]:
Internal error: panic interface conversion: interface {} is nil, not []types.Date: goroutine 3788 [running]:
```

**Review by JinHai-CN — APPROVED** [review]

**Pull request merged**